src/libraries/System.Text.Json/src/System/Text/Json/Document/JsonProperty.cs

```diff
@@ -117,7 +118,46 @@ public void WriteTo(Utf8JsonWriter writer)
                 ThrowHelper.ThrowArgumentNullException(nameof(writer));
             }
 
-            writer.WritePropertyName(Name);
+            if (_name is null)
+            {
+                ReadOnlySpan<byte> rawName = Value.GetRawPropertyNameAsUtf8Span();
+                int firstBackSlashIndex = rawName.IndexOf(JsonConstants.BackSlash);
```

Comment:
Ideally, all this unescaping logic should be encapsulated inside `JsonDocument`/`JsonElement`. Perhaps replace `JsonElement.GetRawPropertyNameAsUtf8Span()` with something like `JsonElement.WritePropertyName(Utf8JsonWriter);`?